microsoft-edge/progressive-web-apps-chromium/how-to/widgets.md

```diff
@@ -0,0 +1,389 @@
+
+Here is an example of what the `data` URL might return: 
+
+```json
+{
+  "name": "Amelie Garner",
+  "meetingNb": 3
+}
+```
+
+
+<!-- ====================================================================== -->
+## Define Widget actions
+
+**OPEN QUESTIONS: Are actions supported yet? How do actions from the manifest work with actions from the ac template? Maybe using the data.tag field in the ac action?**
+
+If you want your Widget to let users perform tasks, use the `actions` field in your Widget definition, and define a template that supports actions.
+
+The `actions` field is an array of action definitions. Here is an example:
+
+```json
+{
+  "name": "Agenda",
+  "description": "Your day, at a glance",
+  "tag": "agenda",
+  "data": "/widgets/data/agenda",
+  "type": "application/json",
+  "ms_ac_template": "/widgets/template/agenda_template.ac.json",
+  "actions": [
+    {
+      "action": "create-meeting",
+      "title": "Create a new meeting"
+    }
+  ]
+}
+```
+
+Use the `action` field to give the action a name. This name will be used to handle the action in your service worker. The `title` field is presented to the user.
+
+To actually trigger the action from the Widget, your template needs to support it. Here is an example of an action defined in a custom Adaptive Cards template:
+
+```json
+{
+  "type": "AdaptiveCard",
+  "body": [
+    {
+      "type": "TextBlock",
+      "size": "Medium",
+      "text": "Hello ${name}"
+    },
+    {
+      "type": "TextBlock",
+      "text": "You have ${meetingNb} meetings today"
+    }
+  ],
+  "actions": [
+    {
+      "type": "Action.Execute",
+      "title": "Create a new meeting",
+      "data": {
+        "tag": "create-meeting"
+      }
+    }
+  ],
+  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+  "version": "1.5"
+}
+```
+
+<!-- ====================================================================== -->
+## Handle Widget actions
+
+Once a user executes an action defined in a Widget, an event is triggered in the PWA's service worker. To handle the user action, listen to the `widgetclick` event in your service worker code:
+
+```javascript
+self.addEventListener('widgetclick', event => {
+  const action = event.action;
+
+  if (action === "create-meeting") {
+    // Application logic code to create a meeting.
+  }
+});
+```
+
+See the [Service Worker API reference](#service-worker-api-reference) below for more information about the `widgetclick` event and what information you can access from it.
+
+
+<!-- ====================================================================== -->
+## Define Widget settings
+
+**OPEN QUESTIONS: Are settings supported yet? If so, are all types implemented?**
```

Comment:
Widget settings will not be supported in SV2, but we do plan on supporting them post-SV2